libs/entitlement/src/main/java/org/elasticsearch/entitlement/runtime/policy/entitlements/FilesEntitlement.java

```diff
@@ -225,71 +221,39 @@ public FileData withPlatform(Platform platform) {
         }
     }
 
-    private record PathSettingFileData(String setting, Mode mode, boolean ignoreUrl, Platform platform, boolean exclusive)
+    private record ConfigPathSettingFileData(String setting, Mode mode, boolean ignoreUrl, Platform platform, boolean exclusive)
         implements
             FileData {
 
         @Override
-        public PathSettingFileData withExclusive(boolean exclusive) {
-            return new PathSettingFileData(setting, mode, ignoreUrl, platform, exclusive);
+        public ConfigPathSettingFileData withExclusive(boolean exclusive) {
+            return new ConfigPathSettingFileData(setting, mode, ignoreUrl, platform, exclusive);
         }
 
         @Override
         public Stream<Path> resolvePaths(PathLookup pathLookup) {
-            return resolvePathSettings(pathLookup, setting, ignoreUrl);
-        }
-
-        @Override
-        public FileData withPlatform(Platform platform) {
-            if (platform == platform()) {
-                return this;
+            Stream<String> result;
+            if (setting.contains("*")) {
```

Comment:
I'm wondering why we need this asterisk check. Does the `settingGlobResolver` path do the wrong thing if the setting contains no asterisk?

Reply:
Yes, the setting glob resolver only knows how to deal with glob.

Reply:
But a glob with no asterisks is still a glob right? Even if it's trivial.

Reply:
It's just a way to understand if we want to call the "regular" Settings resolved (which maps to `Settings#get`) or the glob resolver (which maps to `Settings#getGlobValues`). The latter throws if you don't pass down something that contains a `".*."`
Btw I think we should match that here, and look for `".*."`, not just `*`.